packages/webpack/README.md

```diff
@@ -47,6 +48,40 @@ module.exports = {
 
 One important thing to note when using the LavaMoat plugin is that it disables the `concatenateModules` optimization in webpack. This is because concatenation won't work with wrapped modules.
 
+### Using static shims
+
+Static shims are a way to include additional code in the runtime chunk before LavaMoat starts.
+
+> [!WARNING]
+> Shims cannot use import or require, they must be standalone scripts.
+
+```js
+const LavaMoatPlugin = require('@lavamoat/webpack')
+const path = require('path')
+
+module.exports = {
+  plugins: [
+    new LavaMoatPlugin({
+      staticShims_experimental: [
+        'package-name', 
+        path.join(__dirname, './local/file.js')
+      ]
+    }),
+  ],
+}
+```
+
+The static shims are executed in order before any other code runs.
+A shim can also run code between the repair and harden phases of SES lockdown. To do that, assign a synchronous function to LOCKDOWN_SHIM variable that exists in the scope of the shim.
+It's the only way to polyfill functionality on intrinsics.
+
+```js
+// lockdown-shim.js
+LOCKDOWN_SHIM = () => {
+  Promise.withResolvers = ...
+}
```

Comment:
I'd appreciate comments on the design of this.

I could handle CJS style export instead, but tried to do something else so that I don't give people an idea that the module system might be working where it's not.

I might have put too much effort into making sure that the array of shims is lexically unavailable to the shim. As a result, if you redefine `LOCKDOWN_SHIM` instead of assigning to it, you fail to register a shim.
Is that a problem? I could fix that.

Reply:
By redefine you mean assigning a second time, or using `Obj.defineProps`?

IMO this is fine to get in front of MM and see what they think. If they complain about the API, then we can change it. _It doesn't have to be perfect the first time_ is what I'm saying.  Iterate!

Reply:
Maybe name the variable `PRE_LOCKDOWN_SHIM`?

The API is a bit strange, but it will work!

Reply:
>  get in front of MM and see what they think
They need shims before lockdown, not inbetween lockdown and are unlikely to have the need for those. We need to look at prospective users here :D 

> Maybe name the variable PRE_LOCKDOWN_SHIM?
the top-level code runs before lockdown. the function assigned runs between `repair` and `harden` stages of lockdown. PRE_LOCKDOWN sounds more like the former to me.